packages/xo-web/src/common/utils.js

```diff
@@ -371,13 +370,21 @@ export const resolveResourceSets = resourceSets =>
 //   id: 42,
 // }, 32)
 //
-// newString === 'foo_COPY_foo_42_32'
+// newString === 'foo_COPY_foo_42_32%'
 // ```
 export function buildTemplate(pattern, rules) {
-  const regExp = new RegExp(join(map(keys(rules), escapeRegExp), '|'), 'g')
+  const rulesWithTheirEscape = { '\\\\': '\\' }
+  const escapedSyntaxes = [escapeRegExp('\\\\')]
```

Comment:
`escapedSyntaxes` → `matches`?

And what do you think about not calling `escapeRegExp` everytime but instead calling it later:
```js
... = new RegExp(matches.map(escapeRegExp).join('|'), 'g')
```
It would add a (tiny) loop but I think it would improve readability.

Reply:
Why not :+1: